hail/src/main/scala/is/hail/methods/LocalLDPrune.scala

```diff
@@ -344,7 +366,8 @@ object LocalLDPrune {
       }
     })
 
-    new Table(hc = mt.hc, crdd = sitesOnly.crdd, signature = tableType.rowType, key = tableType.key)
+    TableValue(tableType, BroadcastRow.empty(mv.sparkContext),
```

Comment:
Could you help me understand why we used to, but no longer wrap this in `TableLiteral` (e.g new Table( .. ) ).

Looking at the implementation of TableIR, Table, and TableValue, my first guess is that we no longer need a copy trait/method.

Haven't looked into this very much, so I may am likely missing something obvious.

Reply:
Bit of history: the user-level interface (e.g. Table, MatrixTable) were explicit in Scala and the Python interface was a simple wrapper around the corresponding Scala classes.  As the Python interface became richer and more Pythonic, we decided to abandon the user-level interface in Scala.  That means Scala Table and MatrixTable are on the chopping block (and can nearly be deleted now).  Thus, things that used Table either need to use IR (Table is essentially around a TableIR) or TableValue (the result of evaluating a table).  To turn a TableValue back into a Table, you need to wrap in a TableLiteral to make it an IR.

So this code changed to return a TableValue instead of a Tale, so it no longer needs to wrap the value in a table/literal.

Reply:
In particular, the (temporary) apply wrapper that returns a Table was added to make the existing Scala tests compile, but those should probably be moved to Python and the wrapper should be deleted.

Reply:
Got it thanks!